gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -788,17 +475,10 @@ private static Set<SourceEntity> getSourceEntities(String response) {
     return result;
   }
 
-  /**
-   * Context for probing the table for row counts of a time range
-   */
-  @RequiredArgsConstructor
-  private static class TableCountProbingContext {
-    private final SalesforceConnector connector;
-    private final String entity;
-    private final String watermarkColumn;
-    private final int bucketSizeLimit;
-    private final int probeLimit;
-
-    private int probeCount = 0;
+  SalesforceConnector getConnector(State state) {
```

Comment:
Can we ensure that this function remains protected and not default protected to ensure the same access level for child classes that may extend SalesforceSource?